### PR TITLE
Add calendar feature for multi , range and single selection

### PR DESCRIPTION
**Pull request by karkipy**
@syaau , updates for year selection,

### CODE REVIEW

File: packages/Calendar/src/index.js [hunk 1]
```diff
@@ -16,9 +18,11 @@ const styles = StyleSheet.create({
 });
 
 function getMonthCount({ width }, value) {
-  const date = Array.isArray(value) ? value[0] : value;
-  return width > 400
-    ? [date, new Date(date.getFullYear(), date.getMonth() + 1)] : [date];
+  if (Array.isArray(value)) {
```
Comment (syaau) [inline packages/Calendar/src/index.js]: Isn't this always true ?

**Comment by syaau** [comment]
Too many ifs and then a `switch`. What you are looking for is a navigation pattern.

File: packages/Calendar/src/index.js (hunk 1 above)
Comment (karkipy) [reply packages/Calendar/src/index.js]: no , at first the value is a simple date object

**Comment by karkipy** [comment]
> Too many ifs and then a `switch`. What you are looking for is a navigation pattern.

I used switch case to remove if condition's

File: packages/Calendar/src/MonthView/MonthView.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+function MonthView({ setView }: Props) {
+  const [months, setMonths] = useContext(CalendarContext);
+  const handleLayout = useCallback((e) => {
+    const { layout } = e.nativeEvent;
+    setMonths(mnths => getMonthCount(layout, mnths));
+  }, []);
+
+  const shift = useCallback((val) => {
+    setMonths(mnths => mnths.map(d => new Date(
```
Comment (syaau) [inline packages/Calendar/src/MonthView/MonthView.js]: My instinct tells me this setMonths cannot be used from context.

Comment (karkipy) [reply packages/Calendar/src/MonthView/MonthView.js]: @syaau , but the changes are being reflected

Comment (syaau) [reply packages/Calendar/src/MonthView/MonthView.js]: When you scroll over to select a month from different year, you will end up on the same page in the CalendarView. Given that you have implemented the cancel button.

File: packages/Calendar/src/DayView/Day.js [hunk 3]
```diff
@@ -55,7 +49,6 @@ function Day({
   const dateObj = new Date(date);
   const isCurrentMonth = currentMonth === dateObj.getMonth();
   const isToday = dateObj.toDateString() === new Date().toDateString();
-
   return (
     <TouchableOpacity
       style={[styles.container, borderStyle || { borderColor: 'transparent' }]}
```
Comment (syaau) [inline packages/Calendar/src/DayView/Day.js]: The transparent borderColor could be part of styles.container

File: packages/Calendar/src/DayView/Month.js [hunk 4]
```diff
@@ -3,22 +3,18 @@ import React from 'react';
...
 const WEEK_DIFF = 7 * 86400 * 1000;
 
 function getStartOfMonth(date) {
   const first = new Date(date.getFullYear(), date.getMonth(), 1);
   return first.getTime() - first.getDay() * 86400 * 1000;
 }
 
-
-function Month({ date, ...other }: Props) {
+function Month({ date, view, setView, ...other }: Props) {
```
Comment (syaau) [inline packages/Calendar/src/DayView/Month.js]: What's view and setView doing here ?

File: packages/Calendar/src/DayView/Day.js (hunk 3 above)
Comment (karkipy) [reply packages/Calendar/src/DayView/Day.js]: that creates an issue in android

File: packages/Calendar/src/MonthView/MonthView.js (hunk 2 above)
Comment (karkipy) [reply packages/Calendar/src/MonthView/MonthView.js]: Screenshot_1555473059
Screenshot_1555473075

the view are different but clear selection still resides, shall I remove clear selection ??

File: packages/Calendar/src/Calendar/Day.js [hunk 5]
```diff
@@ -1,26 +1,21 @@
...
     paddingHorizontal: 3,
     paddingVertical: 3,
-    borderLeftWidth: 1,
-    borderRightWidth: 1,
-    borderTopWidth: 1,
-    borderBottomWidth: 1,
-    margin: -1,
+    borderWidth: StyleSheet.hairlineWidth,
+    margin: -StyleSheet.hairlineWidth,
+    borderLeftColor: 'transparent',
```
Comment (syaau) [inline packages/Calendar/src/Calendar/Day.js]: May be you would want to put a comment here. As far as I reckon, this has something to do with android inconsistency

Comment (karkipy) [reply packages/Calendar/src/Calendar/Day.js]: huss

File: packages/Calendar/src/Calendar/CalendarView.js [hunk 6]
```diff
@@ -30,10 +30,19 @@ type Props = {
   setValue: number => void,
   value: ?Date,
   setView: React.Node => void,
+  type: string,
```
Comment (syaau) [inline packages/Calendar/src/Calendar/CalendarView.js]: If you know the types use those names:
`'multi', 'range', 'single'` and default props appropriately.

**Pull request merged**